lib/Service/ApiService.php

```diff
@@ -67,4 +81,35 @@ public function sendFile(int $fileId): void {
 			],
 		);
 	}
+
+	public function searchMessages(string $userId, string $term, int $offset = 0, int $limit = 10): array {
+		$arguments = [
+			'format' => 'json',
+			'query' => '*' . $term . '*' ,
+		];
+
+		$paperlessURL = rtrim($this->config->url, '/') . '/api/documents/?' . http_build_query($arguments);
+		$result = $this->client->get($paperlessURL,
```

Comment:
```suggestion
		$result = $this->client->get($this->config->url . '/api/documents/?' . http_build_query($arguments),
```
Just do it like in the `sendFile()` method. Trimming is not necessary

Reply:
Just my two cents here. I think users usually input the URL including a trailing slash, especially when they copy-paste them from documentation or browser address bars. This was also one of the issues I faced when testing the code, as I didn't realize that I have a trailing slash behind the URL entered. So, this approach will reduce the chances of errors caused by duplicate slashes or incorrect URLs. Let me know your final decision on which approach you'd like to go with.